### DISCUSSION TITLE
OpenSSL 30 vs OpenSSL 102

### DISCUSSION THREAD

**Discussion by ChauhanPriyanka** [openssl/openssl#21920] [Q&A] [answered]
Is there any way to find out certificate is generated using which version of openssl. 
Either from openssl102 apis or openssl30 apis or any other openssl version

**Comment by paulidale** [comment]
No.

**Comment by t8m** [comment, marked as answer]
In general, no. However there might be slight changes in defaults and/or supported features that could give some rough hints as for which version of OpenSSL was used to generate a certificate. On the other hand it is definitely possible to create the same certificate using 1.0.2 as with 3.x if you limit the requirements to things that were already supported in 1.0.2.

**Comment by ChauhanPriyanka** [comment]
HI @t8m ,
thank you for the response, could you please point out to what are the slight changes in defaults and/or supported features that could give some rough hints.
Thanks,
Priyanka

**Reply by t8m** [reply]
Unfortunately I do not have capacity for this right now.

**Comment by mspncp** [comment]
> could you please point out to what are the slight changes in defaults and/or supported features that could give some rough hints.

The [NEWS](https://github.com/openssl/openssl/blob/master/NEWS.md) file in the source tree is a good starting point. It contains a condensed list of changes between the versions.  For example, a brief text search for "x509 cert" found the following entry:

> **[Major changes between OpenSSL 1.1.1 and OpenSSL 3.0.0 [7 sep 2021]](https://github.com/openssl/openssl/blob/master/NEWS.md#major-changes-between-openssl-111-and-openssl-300-7-sep-2021)**
>
> * X509 certificates signed using SHA1 are no longer allowed at security level 1 or higher. The default security level for TLS is 1, so certificates signed using SHA1 are by default no longer trusted to authenticate servers or clients.

If that's not enough, the [CHANGES](https://github.com/openssl/openssl/blob/master/CHANGES.md) file contains more detailed information.
